File: scala/leap/src/main/scala/Leap.scala
object Leap {
  def leapYear(year: Int): Boolean = {
    divisibleBy(year, 4) && !divisibleBy(year, 100) || divisibleBy(year, 400)
  }

  private def divisibleBy(dividend: Int, divisor: Int): Boolean = {
    dividend % divisor == 0
  }
}

File: csharp/leap/Leap.cs
public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        return DivisibleBy(year, 4) &&
            (!DivisibleBy(year, 100) || DivisibleBy(year, 400));
    }

    private static bool DivisibleBy(int dividend, int divisor) {
        return dividend % divisor == 0;
    }
}